pkg/kubelet/certificate/bootstrap/bootstrap_test.go

```diff
@@ -34,9 +35,263 @@ import (
+	copyFile("./testdata/mycertinvalid.crt", fileDir+"/mycertinvalid.crt")
+	copyFile("./testdata/mycertvalid.crt", fileDir+"/mycertvalid.crt")
+	copyFile("./testdata/mycertinvalid.key", fileDir+"/mycertinvalid.key")
+	copyFile("./testdata/mycertvalid.key", fileDir+"/mycertvalid.key")
+	testDataValid := []byte(`
+apiVersion: v1
+kind: Config
+clusters:
+- cluster:
+    certificate-authority: ca-a.crt
+    server: https://cluster-a.com
+  name: cluster-a
+- cluster:
+    server: https://cluster-b.com
+  name: cluster-b
+contexts:
+- context:
+    cluster: cluster-a
+    namespace: ns-a
+    user: user-a
+  name: context-a
+- context:
+    cluster: cluster-b
+    namespace: ns-b
+    user: user-b
+  name: context-b
+current-context: context-b
+users:
+- name: user-a
+  user:
+    client-certificate: mycertvalid.crt
+    client-key: mycertvalid.key
+- name: user-b
+  user:
+    client-certificate: mycertvalid.crt
+    client-key: mycertvalid.key
+
+`)
+	filevalid, err := ioutil.TempFile(fileDir, "kubeconfigvalid")
+	if err != nil {
+		t.Fatal(err)
+	}
+	ioutil.WriteFile(filevalid.Name(), testDataValid, os.FileMode(0755))
+
+	testDataInvalid := []byte(`
+apiVersion: v1
+kind: Config
+clusters:
+- cluster:
+    certificate-authority: ca-a.crt
+    server: https://cluster-a.com
+  name: cluster-a
+- cluster:
+    server: https://cluster-b.com
+  name: cluster-b
+contexts:
+- context:
+    cluster: cluster-a
+    namespace: ns-a
+    user: user-a
+  name: context-a
+- context:
+    cluster: cluster-b
+    namespace: ns-b
+    user: user-b
+  name: context-b
+current-context: context-b
+users:
+- name: user-a
+  user:
+    client-certificate: mycertinvalid.crt
+    client-key: mycertinvalid.key
+- name: user-b
+  user:
+    client-certificate: mycertinvalid.crt
+    client-key: mycertinvalid.key
+
+`)
+	fileinvalid, err := ioutil.TempFile(fileDir, "kubeconfiginvalid")
+	if err != nil {
+		t.Fatal(err)
+	}
+	ioutil.WriteFile(fileinvalid.Name(), testDataInvalid, os.FileMode(0755))
+
+	testDatabootstrap := []byte(`
+apiVersion: v1
+kind: Config
+clusters:
+- cluster:
+    certificate-authority: ca-a.crt
+    server: https://cluster-a.com
+  name: cluster-a
+- cluster:
+    server: https://cluster-b.com
+  name: cluster-b
+contexts:
+- context:
+    cluster: cluster-a
+    namespace: ns-a
+    user: user-a
+  name: context-a
+- context:
+    cluster: cluster-b
+    namespace: ns-b
+    user: user-b
+  name: context-b
+current-context: context-b
+users:
+- name: user-a
+  user:
+   token: mytoken-b
+- name: user-b
+  user:
+   token: mytoken-b
+`)
+	fileboot, err := ioutil.TempFile(fileDir, "kubeconfig")
+	if err != nil {
+		t.Fatal(err)
+	}
+	ioutil.WriteFile(fileboot.Name(), testDatabootstrap, os.FileMode(0755))
+
+	dir, err := ioutil.TempDir(fileDir, "k8s-test-certstore-current")
+	if err != nil {
+		t.Fatalf("Unable to create the test directory %q: %v", dir, err)
+	}
+	defer func() {
+		if err := os.RemoveAll(dir); err != nil {
```

Comment:
no need to delete dir, the parent dir is already getting cleaned up on exit, right?

Reply:
Thanks for your guidance.